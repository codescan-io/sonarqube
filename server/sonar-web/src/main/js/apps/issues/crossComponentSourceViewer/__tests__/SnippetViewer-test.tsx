/*
 * SonarQube
 * Copyright (C) 2009-2023 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import { range } from 'lodash';
import * as React from 'react';
import { mockSourceLine, mockSourceViewerFile } from '../../../../helpers/mocks/sources';
import { mockIssue } from '../../../../helpers/testMocks';
import SnippetViewer from '../SnippetViewer';

jest.mock('../../../../helpers/scrolling', () => ({
  scrollHorizontally: jest.fn(),
}));

beforeEach(() => {
  jest.clearAllMocks();
});

it('should render correctly', () => {
  const snippet = range(5, 8).map((line) => mockSourceLine({ line }));
  const wrapper = shallowRender({
    snippet,
  });

  expect(wrapper).toMatchSnapshot();
});

it('should render correctly with no SCM', () => {
  const snippet = range(5, 8).map((line) => mockSourceLine({ line }));
  const wrapper = shallowRender({
    displaySCM: false,
    snippet,
  });

  expect(wrapper).toMatchSnapshot();
});

it('should render additional child in line', () => {
  const sourceline = mockSourceLine({ line: 42 });

  const child = <div>child</div>;
  const renderAdditionalChildInLine = jest.fn().mockReturnValue(child);
  const wrapper = shallowRender({ renderAdditionalChildInLine, snippet: [sourceline] });

  wrapper.instance().renderLine({
    displayDuplications: false,
    index: 1,
    issueLocations: [],
    line: sourceline,
    snippet: [sourceline],
    symbols: [],
    verticalBuffer: 5,
  });

  expect(renderAdditionalChildInLine).toHaveBeenCalledWith(sourceline);
});

it('should render correctly when at the top of the file', () => {
  const snippet = range(1, 8).map((line) => mockSourceLine({ line }));
  const wrapper = shallowRender({
    snippet,
  });

  expect(wrapper).toMatchSnapshot();
});

it('should render correctly when at the bottom of the file', () => {
  const component = mockSourceViewerFile('foo/bar.ts', 'my-project', { measures: { lines: '14' } });
  const snippet = range(10, 14).map((line) => mockSourceLine({ line }));
  const wrapper = shallowRender({
    component,
    snippet,
  });

  expect(wrapper).toMatchSnapshot();
});

it('should correctly handle expansion', () => {
  const snippet = range(5, 8).map((line) => mockSourceLine({ line }));
  const expandBlock = jest.fn(() => Promise.resolve());

  const wrapper = shallowRender({
    expandBlock,
    index: 2,
    snippet,
  });

  wrapper.find('.expand-block-above button').first().simulate('click');
  expect(expandBlock).toHaveBeenCalledWith(2, 'up');

  wrapper.find('.expand-block-below button').first().simulate('click');
  expect(expandBlock).toHaveBeenCalledWith(2, 'down');
});

function shallowRender(props: Partial<SnippetViewer['props']> = {}) {
  return shallow<SnippetViewer>(
    <SnippetViewer
      component={mockSourceViewerFile()}
      duplications={undefined}
      duplicationsByLine={undefined}
      expandBlock={jest.fn()}
      handleSymbolClick={jest.fn()}
      highlightedLocationMessage={{ index: 0, text: '' }}
      highlightedSymbols={[]}
      index={0}
      issue={mockIssue()}
      lastSnippetOfLastGroup={false}
      loadDuplications={jest.fn()}
      locations={[]}
      locationsByLine={{}}
      onLocationSelect={jest.fn()}
      renderDuplicationPopup={jest.fn()}
      snippet={[]}
      {...props}
    />
  );
}
