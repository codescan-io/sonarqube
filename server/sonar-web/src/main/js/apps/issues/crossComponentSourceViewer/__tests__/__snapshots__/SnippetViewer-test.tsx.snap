// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="source-viewer-code snippet"
>
  <div>
    <div
      className="expand-block expand-block-above"
    >
      <button
        aria-label="source_viewer.expand_above"
        onClick={[Function]}
        type="button"
      >
        <ExpandSnippetIcon />
      </button>
    </div>
    <table
      className="source-table expand-up expand-down"
    >
      <tbody>
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={5}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="5"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 5,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={5}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="6"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 6,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          previousLine={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 5,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={5}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="7"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 7,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          previousLine={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 6,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
      </tbody>
    </table>
    <div
      className="expand-block expand-block-below"
    >
      <button
        aria-label="source_viewer.expand_below"
        onClick={[Function]}
        type="button"
      >
        <ExpandSnippetIcon />
      </button>
    </div>
  </div>
</div>
`;

exports[`should render correctly when at the bottom of the file 1`] = `
<div
  className="source-viewer-code snippet"
>
  <div>
    <div
      className="expand-block expand-block-above"
    >
      <button
        aria-label="source_viewer.expand_above"
        onClick={[Function]}
        type="button"
      >
        <ExpandSnippetIcon />
      </button>
    </div>
    <table
      className="source-table expand-up expand-down"
    >
      <tbody>
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={10}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="10"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 10,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={10}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="11"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 11,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          previousLine={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 10,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={10}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="12"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 12,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          previousLine={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 11,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={10}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="13"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 13,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          previousLine={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 12,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
      </tbody>
    </table>
    <div
      className="expand-block expand-block-below"
    >
      <button
        aria-label="source_viewer.expand_below"
        onClick={[Function]}
        type="button"
      >
        <ExpandSnippetIcon />
      </button>
    </div>
  </div>
</div>
`;

exports[`should render correctly when at the top of the file 1`] = `
<div
  className="source-viewer-code snippet"
>
  <div>
    <table
      className="source-table expand-down"
    >
      <tbody>
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={1}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="1"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 1,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={1}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="2"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 2,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          previousLine={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 1,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={1}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="3"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 3,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          previousLine={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 2,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={1}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="4"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 4,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          previousLine={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 3,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={1}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="5"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 5,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          previousLine={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 4,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={1}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="6"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 6,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          previousLine={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 5,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={1}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="7"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 7,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          previousLine={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 6,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
      </tbody>
    </table>
    <div
      className="expand-block expand-block-below"
    >
      <button
        aria-label="source_viewer.expand_below"
        onClick={[Function]}
        type="button"
      >
        <ExpandSnippetIcon />
      </button>
    </div>
  </div>
</div>
`;

exports[`should render correctly with no SCM 1`] = `
<div
  className="source-viewer-code snippet"
>
  <div>
    <div
      className="expand-block expand-block-above"
    >
      <button
        aria-label="source_viewer.expand_above"
        onClick={[Function]}
        type="button"
      >
        <ExpandSnippetIcon />
      </button>
    </div>
    <table
      className="source-table expand-up expand-down"
    >
      <tbody>
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          displaySCM={false}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={5}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="5"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 5,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          displaySCM={false}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={5}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="6"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 6,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          previousLine={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 5,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
        <Line
          displayCoverage={true}
          displayDuplications={false}
          displayIssues={false}
          displayLocationMarkers={true}
          displaySCM={false}
          duplications={[]}
          duplicationsCount={0}
          firstLineNumber={5}
          highlighted={false}
          issueLocations={[]}
          issues={[]}
          key="7"
          last={false}
          line={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 7,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          loadDuplications={[MockFunction]}
          onIssueSelect={[Function]}
          onIssueUnselect={[Function]}
          onIssuesClose={[Function]}
          onIssuesOpen={[Function]}
          onLocationSelect={[MockFunction]}
          onSymbolClick={[MockFunction]}
          openIssues={false}
          previousLine={
            {
              "code": "<span class="k">import</span> java.util.<span class="sym-9 sym">ArrayList</span>;",
              "coverageStatus": "covered",
              "coveredConditions": 2,
              "duplicated": false,
              "isNew": true,
              "line": 6,
              "scmAuthor": "simon.brandhof@sonarsource.com",
              "scmDate": "2018-12-11T10:48:39+0100",
              "scmRevision": "80f564becc0c0a1c9abaa006eca83a4fd278c3f0",
            }
          }
          renderDuplicationPopup={[MockFunction]}
          secondaryIssueLocations={[]}
          verticalBuffer={0}
        />
      </tbody>
    </table>
    <div
      className="expand-block expand-block-below"
    >
      <button
        aria-label="source_viewer.expand_below"
        onClick={[Function]}
        type="button"
      >
        <ExpandSnippetIcon />
      </button>
    </div>
  </div>
</div>
`;
